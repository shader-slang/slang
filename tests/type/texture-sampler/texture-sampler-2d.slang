//TEST(compute, vulkan):COMPARE_COMPUTE_EX:-vk -compute -shaderobj

//TEST_INPUT: TextureSampler2D(size=4, content=one):name t2D
Sampler2D<float> t2D;

//TEST_INPUT: ubuffer(data=[0 0 0 0], stride=4):out,name outputBuffer
RWStructuredBuffer<float> outputBuffer;

[numthreads(4, 1, 1)]
void computeMain(uint3 dispatchThreadID : SV_DispatchThreadID)
{
    int idx = dispatchThreadID.x;
    float u = idx * (1.0f / 4);
    
    float val = 0.0f;
   
    val += t2D.SampleLevel(float2(u, u), 0);

    outputBuffer[idx] = val;
}
