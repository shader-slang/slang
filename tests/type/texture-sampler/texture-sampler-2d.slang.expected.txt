type: float
1.0
1.0
1.0
1.0
